ocs_ci/ocs/ripsaw.py

```diff
@@ -0,0 +1,125 @@
+
+"""
+RipSaw Class to run various workloads and scale tests
+"""
+import logging
+import tempfile
+
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs import constants
+from subprocess import run, CalledProcessError
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.utility import templating
+
+log = logging.getLogger(__name__)
+
+
+class RipSaw(object):
+    """
+      Workload operation using RipSaw
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        Args:
+            kwargs (dict):
+                Following kwargs are valid
+                repo: Ripsaw repo to used - a github link
+                branch: branch to use from the repo
+                namespace: namespace for the operator
+
+        Example Usage:
+            r1 = RipSaw()
+            r1.apply_crd(crd='ripsaw_v1alpha1_ripsaw_crd.yaml')
+            # use oc apply to apply custom modified bench
+            my_custom_bench = my_custom_bench.yaml
+            run_cmd('oc apply -f my_custom_bench')
+        """
+        self.args = kwargs
+        self.repo = self.args.get('repo', 'https://github.com/cloud-bulldozer/ripsaw')
+        self.branch = self.args.get('branch', 'master')
+        self.namespace = self.args.get('namespace', 'ripsaw')
+        self.pgsql_is_setup = False
+        self.ocp = OCP()
+        self.ns_obj = OCP(kind='namespace')
+        self._create_namespace()
+        self._clone_ripsaw()
+
+    def _create_namespace(self):
+        """
+        create namespace for RipSaw
+        """
+        self.ocp.new_project(self.namespace)
+
+    def _clone_ripsaw(self):
+        """
+        clone the ripaw repo
+        """
+        self.dir = tempfile.mkdtemp(prefix='ripsaw_')
+        try:
+            log.info(f'cloning ripsaw in {self.dir}')
+            git_clone_cmd = f'git clone -b {self.branch} {self.repo} '
+            run(
+                git_clone_cmd,
+                shell=True,
+                cwd=self.dir,
+                check=True
+            )
+            self.crd = 'resources/crd/'
+            self.operator = 'resources/operator.yaml'
+        except (CommandFailed, CalledProcessError)as cf:
+            log.error('Error during cloning of ripsaw repository')
+            raise cf
+
+    def apply_crd(self, crd):
+        """
+        Apply the CRD
+
+        Args:
+            crd (str): Name of file to apply
+        """
+        self.crd = crd
+        self.dir += '/ripsaw'
+        run(f'oc apply -f deploy', shell=True, check=True, cwd=self.dir)
```

Comment:
Please use exec_oc_command from OCP

Reply:
@RazTamir I cannot use that as is without enchanching run_cmd to accept to accept cwd param -> https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/ocp.py#L70
Here we clone the ripsaw and work from temporary folder and the path has to be relative to that temp folder.

I am going to leave it as is and address later on.